packages/metrics_center/lib/src/gcs_lock.dart

```diff
@@ -79,13 +76,28 @@ class GcsLock {
   }
 
   Future<void> _unlock(String lockFileName) async {
-    await _api.objects.delete(_bucketName, lockFileName);
+    Duration waitPeriod = const Duration(milliseconds: 10);
+    bool unlocked = false;
+    // Retry in the case of GCS returning a 504 timeout, but rethrow if unable
+    // to unlock after a certain period of time.
+    while (!unlocked) {
+      try {
+        await _api.objects.delete(_bucketName, lockFileName);
+        unlocked = true;
+      } on DetailedApiRequestError catch (e) {
+        if (e.status == 504 && waitPeriod < _unlockThreshold) {
```

Comment:
I would prefer to skip the status code, we may have other 500+ errors and let's retry on any `DetailedApiRequestError`.

Reply:
I removed the status check and the associated test, so now it will retry on any `DetailedApiRequestError`!